File: public/dsa/searching/insert_position_in_sorted_array/lib/insert_position_in_sorted_array.cpp
#include <bits/stdc++.h>
using namespace std;

int getInsertPosition(vector<int> &arr, int key) {
    return lower_bound(arr.begin(), arr.end(), key) - arr.begin();
}

// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 5};
    int key = 4;
    cout << getInsertPosition(arr, key) << endl;
    return 0;
}
// tests-end

/*output
3
*/

File: public/dsa/searching/insert_position_in_sorted_array/lib/insert_position_in_sorted_array.py


from bisect import bisect_left
from typing import List


class Solution:
    def getInsertPosition(self, arr: List[int], key: int) -> int:
        return bisect_left(arr, key)


# tests-start
arr = [1, 2, 3, 5]
key = 4
sol = Solution()
print(sol.getInsertPosition(arr, key))
# tests-end


'''output
3
'''
